[starchypotatocode] iP

starchypotatocode opened the pull request:
# KaiPro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

KaiPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/starchypotatocode/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

j-lum committed:
Merge pull request #3 from j-lum/inline-javafx-tutorials

Inline JavaFX tutorials

j-lum committed:
Update Gradle tutorial to reflect new checkstyle config

j-lum committed:
Fix typo referring to `HelloWorld`

j-lum committed:
Add images to JavaFX tutorial 3

j-lum committed:
Merge pull request #7 from j-lum/tutorial-fix

Address issues #5 and #6

j-lum committed:
Remove references to DukeStub

j-lum committed:
Merge pull request #8 from j-lum/tutorial-fix

Remove references to DukeStub

j-lum committed:
Change indentation in FXML samples to match AB-3

j-lum committed:
Merge pull request #14 from j-lum/fxml-indent

Change indentation in FXML samples to match AB-3

j-lum committed:
JavaFX tutorial: Support cross-platform JARs

The OpenJFX plugin expects applications to be modular and bundled
with jlink, resulting in fat jars that are not cross-platform. Let's
manually include the required dependencies so that shadow can package
them properly.

j-lum committed:
Merge pull request #35 from j-lum/xplatformjfx

JavaFX tutorial: Support cross-platform JARs

jiachen247 committed:
Remove unnecessary override annotation

Remove extra '@Override' at the start of the code example.

j-lum committed:
Merge pull request #41 from jiachen247/patch-2

Remove unnecessary override annotation

Commit:
Remove generic tutorials

Let's remove generic tutorials from this repo as they are not specific
to this project tempalte

Commit:
Add text-ui-test files

Commit:
Add Gradle support

damithc committed:
README.md: Fix typo of => or

Commit:
Improve runtest.* scripts

* Improve the javac command to work for multiple java files.
* In runtest.bat, delete ACTUAL.TXT only if it exists.

Commit:
README: Update setup instructions to match the latest IDE version

Commit:
README.md: Tweak template text

* Let's replace 'Feature 1' with 'Feature-ABC'
  reason: some students kept the 'Feature 1' in the final UG.
* Let's tweak the template for features and usage
  reasons: to fix grammar, to distinguish between output and outcome

damithc committed:
README: Mention the project language level setting

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
README: Update the JDK version 11 -> 17

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

starchypotatocode committed:
Modifying .gitignore

starchypotatocode committed:
Edited chatbot for distinctness

starchypotatocode committed:
Added echo functionality

starchypotatocode committed:
Replaced echo functionality
with task record/reminder system

starchypotatocode committed:
Fixed typo

starchypotatocode committed:
Refactored task functionality into Task.java
and implemented basic mark/unmark stuff
lastly, basic exception handling is done

starchypotatocode committed:
Added more task variants,
although testing isn't very complete

starchypotatocode committed:
Fixed error handling bugs,
setting up initial testing

starchypotatocode committed:
Bug fixing regarding input validation
and task listing

starchypotatocode committed:
Fixed text-based testing

starchypotatocode committed:
Fully implemented exception handling for now
with actual classes for that purpose too

starchypotatocode committed:
Implemented delete command,
and updated test cases to match

starchypotatocode committed:
Merge branch 'master' of https://github.com/starchypotatocode/ip

starchypotatocode committed:
Fixed merge snafu
implemented delete command with tests

starchypotatocode committed:
Align to coding standard

starchypotatocode committed:
Partial refactoring of logic

Add Parser.java, Storage.java, Ui.java,
and move some of the logic out there

starchypotatocode committed:
Add save/load functionality

starchypotatocode committed:
Merge branch 'branch-Level-7'

starchypotatocode committed:
Clean up imports and redudancies

starchypotatocode committed:
Continue refactoring Parser.java

starchypotatocode committed:
Fix Parser.java bug

starchypotatocode committed:
Separate files into packages

starchypotatocode committed:
Add TaskList.java and refactor

starchypotatocode committed:
Fix git missing files

starchypotatocode committed:
Undo Ui.java changes

starchypotatocode committed:
Add Command.java, ExitCommand.java

starchypotatocode committed:
Add MarkCommand.java, UnmarkCommand.java

starchypotatocode committed:
Add DeleteCommand.java, ListCommmand.java

starchypotatocode committed:
Add CreateDeadlineCommand.java

starchypotatocode committed:
Add CreateToDoCommand.java

starchypotatocode committed:
Add CreateEventCommand.java

starchypotatocode committed:
Refactor out Parser.java into Command.java and extending classes

starchypotatocode committed:
Add InvalidCommand.java and associated logic

starchypotatocode committed:
Fix grammar and unused imports

starchypotatocode committed:
Refactor more logic into Ui.java

starchypotatocode committed:
Fix control flow errors

starchypotatocode committed:
Merge branch 'branch-A-MoreOOP'

starchypotatocode committed:
Force /from, /to and /by to only accept valid dates

starchypotatocode committed:
Configure gradle for project

starchypotatocode committed:
Fix coding style violations

starchypotatocode committed:
Merge branch 'branch-Level-8'

starchypotatocode committed:
Merge branch 'add-gradle-support'

starchypotatocode committed:
Add gradle and checkstyle support

starchypotatocode committed:
Add JUnit functionality and some tests

starchypotatocode committed:
Fix incorrect test

starchypotatocode committed:
Fix storage file creation and add Kai.jar generation

starchypotatocode committed:
Update JavaDoc documentation

starchypotatocode committed:
Fix coding standard issues

starchypotatocode committed:
Further coding standard fixes

starchypotatocode committed:
Add find command and relevant functionality

starchypotatocode committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

starchypotatocode committed:
Fix merge conflicts

starchypotatocode committed:
Fix Ui.java JavaDoc

starchypotatocode committed:
Add KaiPerserTest.java and partial tests

starchypotatocode committed:
Initial integration of GUI code

starchypotatocode committed:
Fix integration bugs

starchypotatocode committed:
Redirect text output to GUI

starchypotatocode committed:
Rename files and references

starchypotatocode committed:
Refactor control flow

starchypotatocode committed:
Replace images

starchypotatocode committed:
Add title to window

kaoxi998533 commented on src/main/java/kai/KaiParser.java:
```diff
@@ -0,0 +1,207 @@
+     * @param taskList the TaskList some of these commands work on
+     * @return the Command to be invoked later on
+     */
+    public Command parseCommand(String input, TaskList taskList) {
```
Should some branches be extracted out to reduce verbosity of code?

starchypotatocode replied:
Wdym? parseExitCommand etc, or more likely parseCreateTaskCommand factored out?
Didn't do it because I think the amount of code will remain the same with more JavaDoc boilerplate, or did you have something else in mind?

kaoxi998533 commented on src/main/java/kai/commands/CreateDeadlineCommand.java:
```diff
@@ -0,0 +1,46 @@
+ * It is immediately implemented as one-use
+ * to avoid potential re-use issues down the line
+ */
+public class CreateDeadlineCommand extends Command {
```
Should class have noun names rather than verbs?

starchypotatocode replied:
~Pretend~ I failed my English Classes: How would an example look like? Thanks!

kaoxi998533 commented on src/main/java/kai/commands/FindCommand.java:
```diff
@@ -0,0 +1,41 @@
+ * whose descriptions match the String that is given
+ */
+public class FindCommand extends Command {
+    private final TaskList taskList;
```
I like how you utilise final fields to prevent unexpected behaviour of class

kaoxi998533 commented on src/main/java/kai/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+     *
+     * @param tasks the list of tasks in the class
+     */
+    public TaskList(ArrayList<Task> tasks) {
```
Please note that it would be more convenient if there were a constructor that takes no argument

starchypotatocode replied:
Fair, I'll just this(new ArrayList<Task>()) this or something I suppose: fix it some other time, thanks.

kaoxi998533 reviewed (commented):
LGTM!
Some minor changes needed.

starchypotatocode committed:
Fix code quality issues

To ensure future programmers have an easier time
trying to understand the code

- Extract Methods (KaiParser.java)
- Avoid changing input args
- Rename classes with nouns
- Fix other linting issues

starchypotatocode committed:
Add asssertions

As others may not know the implicit
assumptions coded into the methods here

starchypotatocode committed:
Merge pull request #1 from starchypotatocode/branch-A-Assertions

Add asssertions

starchypotatocode committed:
Merge branch 'master' into branch-A-CodeQuality

starchypotatocode committed:
Fix broken tests

starchypotatocode committed:
Merge pull request #2 from starchypotatocode/branch-A-CodeQuality

Fix code quality issues

starchypotatocode committed:
Create gradle.yml

starchypotatocode committed:
Add functionality to reject duplicates

starchypotatocode committed:
Fix checkstyle issues

starchypotatocode committed:
Fix comment messages

starchypotatocode committed:
Refactor SLAP violations

starchypotatocode committed:
Use AI to standardise some files

KaiParser.java, Ui.java and Storage.java
had quite a few issues with SLAP and comment messages.
Fed the text into ChatGPT for suggestions,
and then minorly edited incorrect output before committing.

starchypotatocode committed:
Add user interface sample image

starchypotatocode committed:
Add image to README.md

starchypotatocode committed:
Add explanation for List Command

starchypotatocode committed:
Add section for each type of command

starchypotatocode committed:
Fill in user guide

starchypotatocode committed:
Fix grammar and typos
